steam/src/steam.cpp

```diff
@@ -56,19 +56,63 @@ int Steam_OnGameOverlayActivated(lua_State* L, void* data)
 	return 2;
 }
 
+// Function to convert an array of unsigned char to a hexadecimal string
+void unsignedCharArrayToHex(const unsigned char* data, size_t length, char* hexString) {
+	const char hexDigits[] = "0123456789ABCDEF"; // Hexadecimal digit characters
+	// Iterate over each byte in the data array
+	for (size_t i = 0; i < length; ++i) {
+		// Extract high and low nibbles (4-bit segments) of the byte
+		hexString[i * 2] = hexDigits[(data[i] >> 4) & 0x0F]; // High nibble
+		hexString[i * 2 + 1] = hexDigits[data[i] & 0x0F];    // Low nibble
+	}
+	// Null-terminate the string
+	hexString[length * 2] = '\0';
+}
+
+int Steam_OnGetTicketForWebApiResponse(lua_State* L, void* data)
+{
+	GetTicketForWebApiResponse_t* s = (GetTicketForWebApiResponse_t*)data;
+	lua_pushstring(L, "GetTicketForWebApiResponse_t");
+	lua_newtable(L);
+	lua_pushstring(L, "m_hAuthTicket");
+	lua_pushinteger(L, s->m_hAuthTicket);
```

Comment:
The documentation recommends that you call CancelAuthTicket() when you are finished using it. Perhaps CancelAuthTicket(s->m_hAuthTicket) should be called here at the end of the function?

https://partner.steamgames.com/doc/api/ISteamUser#GetAuthTicketForWebAp

Reply:
Or actually, CancelAuthTicket() must probably be called once you are done using it in your client code. Not as I initially suggested.

You can perhaps store the `s->m_hAuthTicket` in `g_CurrentAuthTicket` or something and provide a function to explicitly cancel the ticket and set g_CurrentAuthTicket to 0. Also done automatically SteamUser_Final() or before calling the function again.